### ISSUE TITLE
Punchbots not always using melee appropriately

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/CleverGirl#6]
Reported by Harkonnen in closed-3437 on RT discord:

```
~400 dmg punchbot decided not to melee ~900 dmg punchbot
2020-04-09 01:14:13.706 -  meleeDamageRatio: 0.2453977 = target: 240.4846 / attacker: 59.01437 vs. cap: 1.1
2020-04-09 01:14:13.706 -  potential melee damage too high, skipping melee.
It was still by far the best action it could have taken
```

**Comment by IceRaptor** [comment]
[clever_girl.log.1.zip](https://github.com/IceRaptor/CleverGirl/files/4475986/clever_girl.log.1.zip)

**Comment by IceRaptor** [comment]
Added semantic that rawMelee >= 2x rawRanged (as weapon.DamagePerShot * weapon.ShotsWhenFired) means you're a punchbot harry. And you'll always want to melee regardless of retaliation.

**Comment by IceRaptor** [comment]
Should be fixed in 0.4.5. CBTBE units with `CBTBE_Punch_Is_Physical_Weapon` stat will ignore retaliation. If CBTBE melee not used, calculation is controlled by mod.config.weights.PunchbotDamageMulti.
